README.md

```diff
@@ -1,136 +1,583 @@
+&nbsp;&nbsp;&nbsp;
+<img src="images/check-green.png" width="20px" height="20px" class="check" alt="Supported"> <img src="images/edge.png" width="20px" height="20px"> Edge
+
+**Unsupported:** &nbsp; <img src="images/red-cross.png" width="20px" height="20px" class="check" alt="Unsupported"> <img src="images/ie.png" width="20px" height="20px"> Internet Explorer
+
+## Project files
+
+There are 3 ways to specify the files of a playground project.
+
+### Option 1: Inline scripts
+
+Add one or more `<script type="sample/..." filename="...">` tags as children of
+your `<playground-ide>` or `<playground-project>`.
+
+Be sure to escape closing `</script>` tags within your source as `<&lt;script>`.
+
+Leading whitespace that is common to all lines is automatically removed unless
+the `preserve-whitespace` attribute is present.
 
 ```html
-<playground-ide>
+<playground-project>
   <script type="sample/html" filename="index.html">
-    <script type="module" src="my-element.js">&lt;/script>
-    <h1>Hello World!</h1>
-    <my-element></my-element>
+    <!doctype html>
+    <head>
+      <script type="module" src="javascript.js">&lt;/script>
+      <script type="module" src="typescript.js">&lt;/script>
+      <link rel="stylesheet" href="styles.css">
+    </head>
+    <body>
+      <p>Hello World!</p>
+    </body>
   </script>
-  <script type="sample/js" filename="my-element.js">
-    import {LitElement, html, customElement} from 'lit-element';
 
-    class MyElement extends LitElement { /* ... */ }
-    customElements.define('my-element', MyElement);
+  <script type="sample/js" filename="javascript.js">
+    console.log('hello from javascript');
   </script>
-</playground-ide>
+
+  <script type="sample/ts" filename="typescript.ts">
+    console.log('hello from typescript');
+  </script>
+
+  <script type="sample/css" filename="styles.css">
+    body { color: blue; }
+  </script>
+</playground-project>
 ```
 
-Or define your project in a JSON manifest:
+### Option 2: JSON manifest
+
+Serve a JSON file containing a `files` object, with relative filenames.
 
 ```html
-<playground-ide project-src="./example-1.json"></playground-ide>
+<playground-ide project-src="/path/to/my/project.json"> </playground-ide>
 ```
 
-`example-1.json`:
-
 ```json
 {
   "files": {
     "index.html": {},
-    "my-element.js": {},
-    "my-second-element.js": {}
+    "javascript.js": {},
+    "typescript.ts": {},
+    "styles.css": {}
   }
 }
 ```
 
-## Getting Started
+## Option 3: Files property
 
-Install with npm:
+Directly set the `files` property to an array of files.
 
-```sh
-npm i playground-elements
+```js
+const ide = document.querySelector('playground-ide');
+ide.files = [
+  {
+    name: "index.html",
+    content: "<p>Hello World!</p>",
+  },
+  ...
+],
 ```
 
-Load the component definition:
+If both `project-src` and `files` are set, then the one set most recently has
+precedence. When either are set, inline scripts are ignored.
+
+## Module resolution
```

Comment:
I think it might be good to have a section on TypeScript, mainly to document what the compiler settings are and say that .ts files are imported with .js extensions like how `tsc` works.

Reply:
Done.